source/common/stats/hystrix_stats.cc

```diff
@@ -18,14 +18,27 @@ void HystrixStats::pushNewValue(std::string key, int value){
 
 uint64_t HystrixStats::getRollingValue(std::string cluster_name, std::string stats) {
   std::string key = "cluster." + cluster_name + "." + stats;
-  if (rolling_stats_map_.find(key) != rolling_stats_map_.end())
-    // TODO: counter may be reset during action
-    return rolling_stats_map_[key][current_index_]-
-        rolling_stats_map_[key][(current_index_+1)%num_of_buckets_];
+
+  if (rolling_stats_map_.find(key) != rolling_stats_map_.end()) {
+    // if the counter was reset, the result is negative
+    // better return 0, will be back to normal once one rolling window passes
+    // better idea what to return? could change the algorithm to keep last valid delta,
+    // updated with pushNewValue and kept stable when the update is negative.
+    if (rolling_stats_map_[key][current_index_] <
+        rolling_stats_map_[key][(current_index_+1)%num_of_buckets_]) {
+      return 0;
```

Comment:
Do we have places that 0 is a common value? If so, we need to think on another solution